### PR TITLE
Update pints.plot

### LINKED ISSUE #249: Update examples and plotting functions: Change `chains` to `samples`
Make sure that:

1. All plotting functions that take lists of samples (or 2d arrays) use `samples`, not `chains`
2. Some of the examples use `samples` rather than `chains`

### LINKED ISSUE #263: Plotting functions: Change argument name `true_values`
We're using `values` to mean the values of the time series in most of pints, whereas here we mean the parameters

### LINKED ISSUE #267: Add true parameters option to histogram / trace plots
So need to change the signature from trace(chain1, chain2, ...) to trace([chain1, chain2, ...], reference=123)

Still not sure about the best name for the true parameters, see #263

### PR DESCRIPTION
**Pull request by chonlei**
Update pints.plot and should 
Fix #249 
Fix #263 
Fix #267

### CODE REVIEW

**Comment by chonlei** [comment]
Need to change and update examples before merge.

File: pints/plot.py [hunk 1]
```diff
@@ -168,65 +168,81 @@ def function_between_points(f, point_1, point_2, padding=0.25, evaluations=20):
...
     Returns a `matplotlib` figure object and axes handle.
     """
     import matplotlib.pyplot as plt
 
+    # If we switch to Python3 exclusively, bins and alpha can be keyword-only
+    # arguments
     bins = 40
     alpha = 0.5
-    n_sample, n_param = chain.shape
+    samples = np.array(samples)  # try to handle numpy array only
```
[the remark this answers is not in the document]
Comment (chonlei) [reply pints/plot.py]: Why we want it to be different sizes? (We cannot have `samples.shape()` next line then, if we are handling lists.)

[the remark this answers is not in the document]
Comment (MichaelClerx) [reply pints/plot.py]: Yeah you can't have the shape line, but apart from that...

[the remark this answers is not in the document]
Comment (chonlei) [reply pints/plot.py]: Fixed.

**Comment by chonlei** [comment]
Ready to merge. Rebased onto master and updated all examples that use `pints.plot` including the new ones. This PR should

Close #273 
Close #270 
Close #267 
Close #263 
Close #249

File: pints/plot.py [hunk 2]
```diff
@@ -168,65 +168,105 @@ def function_between_points(f, point_1, point_2, padding=0.25, evaluations=20):
...
+                xmin = np.min(samples_j[:, i])
+                xmax = np.max(samples_j[:, i])
+            else:
+                xmin = np.percentile(samples_j[:, i],
+                                     50 - n_percentiles / 2.)
+                xmax = np.percentile(samples_j[:, i],
+                                     50 + n_percentiles / 2.)
+            xbins = np.linspace(xmin, xmax, bins)
+            axes[i].hist(samples_j[:, i], bins=xbins, alpha=alpha,
+                         label='Samples ' + str(1 + j_list))
```
Comment (MichaelClerx) [inline pints/plot.py]: Maybe something like "List 1", "List 2", etc.?

**Comment by MichaelClerx** [comment]
Looks good @chonlei !
Can you maybe "Squash and merge" so that we have fewer commits? (If you do, make sure to write "Closes #1234 etc. in the new commit message you make).

**Pull request merged**